src/app/public/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.ts

```diff
@@ -0,0 +1,233 @@
+      .subscribe(dataState => {
+        this._dataState = dataState;
+        this.onlyShowSelected = dataState.onlyShowSelected;
+        this.changeDetector.markForCheck();
+      });
+
+    this.dataManagerService.getDataManagerConfigUpdates()
+      .pipe(takeUntil(this._ngUnsubscribe))
+      .subscribe(config => {
+        this.dataManagerConfig = config;
+      });
+   }
+
+   public sortSelected(sortOption: SkyDataManagerSortOption): void {
+    this.dataState.activeSortOption = sortOption;
+    this.dataManagerService.updateDataState(this.dataState, this._source);
+   }
+
+  public onViewChange(viewId: string): void {
+    this.dataManagerService.updateActiveViewId(viewId);
+  }
+
+  public searchApplied(text: string): void {
+    this.dataState.searchText = text;
+    this.dataManagerService.updateDataState(this.dataState, this._source);
+  }
+
+  public filterButtonClicked(): void {
+    const context = new SkyDataManagerFilterModalContext();
+    const filterModal = this.dataManagerConfig && this.dataManagerConfig.filterModalComponent;
+
+    context.filterData = this.dataState && this.dataState.filterData;
+
+    const options: any = {
+      providers: [{ provide: SkyDataManagerFilterModalContext, useValue: context }]
+    };
+
+    if (filterModal) {
+      const modalInstance = this.modalService.open(filterModal, options);
+
+      modalInstance.closed.subscribe((result: SkyModalCloseArgs) => {
+        if (result.reason === 'save') {
+          this.dataState.filterData = result.data;
+          this.dataManagerService.updateDataState(this.dataState, this._source);
+        }
+      });
+    }
+  }
+
+  public openColumnPicker(): void {
+    const context = new SkyDataManagerColumnPickerContext();
+    const viewState = this.dataState.getViewStateById(this.activeView.id);
+    context.columnOptions = this.activeView && this.activeView.columnOptions;
+    context.displayedColumnIds = viewState.displayedColumnIds;
+
+    const options: any = {
+      providers: [{ provide: SkyDataManagerColumnPickerContext, useValue: context }]
+    };
+
+    const modalInstance = this.modalService.open(SkyDataManagerColumnPickerComponent, options);
+
+    modalInstance.closed.subscribe((result: SkyModalCloseArgs) => {
+      if (result.reason === 'save') {
+        const displayedColumnIds = result.data.map((col: SkyDataManagerColumnPickerOption) => col.id);
+
+        viewState.displayedColumnIds = displayedColumnIds;
+        this.dataState = this.dataState.addOrUpdateView(this.activeView.id, viewState);
+        }
+    });
+  }
+
+  public selectAll(): void {
+    this.activeView.onSelectAllClick();
+  }
+
+  public clearAll(): void {
+    this.activeView.onClearAllClick();
+  }
+
+  public onOnlyShowSelected(event: SkyCheckboxChange): void {
+    this.dataState.onlyShowSelected = event.checked;
+    this.dataManagerService.updateDataState(this.dataState, this._source);
+  }
+
+  public ngOnDestroy(): void {
```

Comment:
This is super nitpicky, but would you mind keeping the Angular implemented methods together (`ngOnInit`, `ngOnDestroy`)? We typically keep them at the top of the list of methods.